presto-common/src/main/java/com/facebook/presto/common/block/VariableWidthBlockBuilder.java

```diff
@@ -244,9 +245,12 @@ public BlockBuilder writeLong(long value)
     @Override
     public BlockBuilder writeBytes(Slice source, int sourceIndex, int length)
     {
+        checkValidSliceRange(sourceIndex, length);
+
         if (!initialized) {
             initializeCapacity();
         }
+
         sliceOutput.writeBytes(source, sourceIndex, length);
```

Comment:
Looking through the airlift code it seems to me that writeBytes already does this when it's trying to allocate more space. This seems redundant. I suppose this check will get called before the one in writeBytes, but does that change anything?

Reply:
The existing code doesn't consider the scenario when sourceIndex + length crosses integer limit. This leads to a negative index and ended up with IndexOutOfBoundException. I am addressing that scenario.